tests/functional/adapter/test_cratedb.py

```diff
@@ -0,0 +1,136 @@
+import pytest
+from dbt.tests.adapter.simple_snapshot import common
+
+from dbt.tests.util import check_relations_equal, run_dbt
+
+
+basic_sql = """
+SELECT 42
+"""
+
+rename_table_sql = """
+{% set exists, table_source = get_or_create_relation(database=None, schema=target.schema, identifier="table_source", type="table") %}
+{% set exists, table_target = get_or_create_relation(database=None, schema=target.schema, identifier="table_target", type="table") %}
+
+{# Create table `source`. #}
+{% set sql = get_create_table_as_sql(False, table_source, "SELECT 42 as value") %}
+{% do run_query(sql) %}
+
+{# Rename table to `target`. #}
+{% do adapter.rename_relation(table_source, table_target) %}
+
+-- Dummy statement.
+SELECT TRUE
+"""
+
+
+rename_view_sql = """
+{% set exists, view_source = get_or_create_relation(database=None, schema=target.schema, identifier="view_source", type="view") %}
+{% set exists, view_target = get_or_create_relation(database=None, schema=target.schema, identifier="view_target", type="view") %}
+
+{# Create view `source`. #}
+{% set sql = get_create_view_as_sql(view_source, "SELECT 42 as value") %}
+{% do run_query(sql) %}
+
+{# Rename view to `target`. #}
+{% do adapter.rename_relation(view_source, view_target) %}
+
+-- Dummy statement.
+SELECT TRUE
+"""
+
+
+reset_csv_table = """
+{# Create a random table. #}
+{% set exists, random_table = get_or_create_relation(database=None, schema=target.schema, identifier="random_table", type="table") %}
+{% set sql = get_create_table_as_sql(False, random_table, "SELECT 42 as value") %}
+{% do run_query(sql) %}
+
+{#
+This is a little excerpt from dbt/include/global_project/macros/materializations/seeds/seed.sql
+#}
+{% set full_refresh = (should_full_refresh()) %}
+{% set agate_table = None %}
+{% set sql_ddl = reset_csv_table(model=model, full_refresh=full_refresh, old_relation=random_table, agate_table=agate_table) %}
+{% set sql_dml = load_csv_rows(model, agate_table) %}
+{% set sql = get_csv_sql(sql_ddl, sql_dml) %}
+{% do run_query(sql) %}
+
+SELECT TRUE
+"""
```

Comment:
That's the test case for validating CrateDB's macro override `cratedb__reset_csv_table`. It is being used by the upstream materialization macro `materialization seed, default` in [macros/materializations/seeds/seed.sql](https://github.com/dbt-labs/dbt-adapters/blob/main/dbt/include/global_project/macros/materializations/seeds/seed.sql).

The test case is a bit too verbose for my taste, but I haven't been able to come up with a better one, because of my lack of knowledge and fluency in dbt.

@hlcianfagna: Maybe you have a better and more concise dbt recipe at hand, which (maybe indirectly) tests whether CrateDB's custom override `cratedb__reset_csv_table` is properly in place?

Reply:
I was testing this manually, I had a tiny csv in the `seeds` folder and I was running `dbt seed` twice.

Reply:
Thanks. I added a corresponding software test that invokes `dbt seed` twice per ad768c521ec09d34, but this one apparently does not trigger `reset_csv_table`. So I've kept the previous software test in place for now.